lobster/docs/source/lobster_to_md.lobster

```diff
@@ -0,0 +1,75 @@
+/*
```

Comment:
I still don't understand how this works. This code appears to work on an internal example, but where is the code from which `c_style_cheat_sheet.md` was generated?
Shouldn't this code be set up such that it generates `c_style_cheat_sheet.md`, and maybe call that from the script that generates all documents?

Reply:
Ah, I think there might've been some misunderstanding on my part. I wrote `c_style_cheat_sheet.md` manually, just using the converter on code examples, then copying and pasting them into my editor which made sure they were properly formatted into a table. If you turn word wrapping off on that file, you'll see it's just a formatted table, not the mess word wrapping turns it into haha.

What would be the best way to generate such a file like `c_style_cheat_sheet.md`? Would it be something like my `.doc` syntax that&mdash;instead of HTML&mdash;outputs a markdown file, or something entirely different?

Reply:
Ok, I am just thinking in terms of how this file should be edited in the future. All the `&nbsp;` in particular makes the code unreadable in the `.md`. And there is no easy way to edit this code in this form, or even to paste it into your tool.

Does the escaping even need to be this heavy handed, other than lt/gt?

I would love to have everything in `.md`, but if it becomes harder to edit than `.html`, that's not good. Also, needing a relatively complicated tool that understands most of `.md` doesn't sound great either.

Reply:
Yeah, that's what I've been thinking about too. It's a weirdly specific problem to fix.

As for the characters that **need** to be converted. I'd say the most minimal set would be:
1. `<` and `>` -> `&lt;`, `&gt;`
2. `\n` -> `<br/>` &mdash; Since the code needs to fit in a markdown table row, we need some way to keep the original line breaks intact.
3. `\s` -> `&nbsp;` &mdash; When `<br/>` is used, there needs to be a way to keep the original indentation intact.

As long as the examples are wrapped in `<code>` tags, those should be the only characters that need to be converted (AFAIK).

Reply:
I was looking here: https://pandoc.org/MANUAL.html
Search for `grid_tables`.
Says: "The cells of grid tables may contain arbitrary block elements (multiple paragraphs, code blocks, lists, etc.). " which sounds promising
So in theory multiple lines of a cell could all preceded by 4 spaces to indicate a code block and be formatted correctly?

Reply:
Do you think using these tables is feasible?